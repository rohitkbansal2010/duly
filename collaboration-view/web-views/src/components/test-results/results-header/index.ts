export { ResultsHeader } from './results-header';
