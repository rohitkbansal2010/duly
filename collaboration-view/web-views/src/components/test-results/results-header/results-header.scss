@import "~@styles/typography";
@import "~@styles/colors";

.results {
  &Header {
    display: flex;
    align-items: center;

    &Icon {
      width: 2rem;

      &Wrapper {
        display: flex;
        align-items: center;
        justify-content: center;
        width: 3.75rem;
        height: 3.75rem;
        margin-right: 1.125rem;
        background: $light-blue;
        border-radius: 50%;
      }
    }

    &Titles {
      display: flex;
      flex-direction: column;
      align-items: flex-start;

      &Main {
        @include nexa-extra-bold-highlight;

        font-size: var(--font-size-24px-4px);
        line-height: 1.875rem;
        text-align: left;
      }

      &Additional {
        @include inter-text-xs-bold;
        font-size: var(--font-size-14px-4px);
        line-height: var(--line-height-16px-4px);
      }
    }
  }
}
