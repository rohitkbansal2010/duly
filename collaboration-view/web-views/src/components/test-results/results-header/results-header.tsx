import { testResultsLightBlueIcon } from '@icons';
import { getTestReportDate } from '@utils';

import styles from './results-header.scss';

type ResultsHeaderProps = {
  title?: string;
  effectiveDate?: string;
}

export const ResultsHeader = ({ title = '', effectiveDate = '' }: ResultsHeaderProps) =>
  (
    <div className={styles.resultsHeader}>
      <div className={styles.resultsHeaderIconWrapper}>
        <img
          src={testResultsLightBlueIcon}
          alt="test result icon"
          className={styles.resultsHeaderIcon}
        />
      </div>
      <div className={styles.resultsHeaderTitles}>
        <span className={styles.resultsHeaderTitlesMain}>
          {title}
        </span>
        <span className={styles.resultsHeaderTitlesAdditional}>
          {effectiveDate && getTestReportDate(effectiveDate)}
        </span>
      </div>
    </div>
  );
