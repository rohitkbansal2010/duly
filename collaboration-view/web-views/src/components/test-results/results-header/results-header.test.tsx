import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { testResultsLightBlueIcon } from '@icons';
import { testReportsResultMockData } from '@mock-data';
import { getTestReportDate } from '@utils';

import { ResultsHeader } from './results-header';

describe('ResultsHeader', () => {
  let component: ReturnType<typeof shallow>;

  const {
    title,
    effectiveDate,
  } = testReportsResultMockData;
  
  const mockProps = {
    title,
    effectiveDate,
  };

  beforeEach(() => {
    component = shallow(<ResultsHeader {...mockProps} />);
  });

  it('component should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should contain right data', () => {
    const formattedDate = getTestReportDate(mockProps.effectiveDate);

    expect(component.text().includes(mockProps.title)).toBeTruthy();
    expect(component.text().includes(formattedDate)).toBeTruthy();
    expect(component.findWhere(n => 
      n.type() === 'img' && n.props().src === testResultsLightBlueIcon)).toBeTruthy();
  });

  it('should work with empty data', () => {
    component = shallow(<ResultsHeader />);

    expect(component.text().includes('Invalid Date')).toBeFalsy();
    expect(component.text().includes('undefined')).toBeFalsy();
    expect(toJson(component)).toMatchSnapshot();
  });
});
