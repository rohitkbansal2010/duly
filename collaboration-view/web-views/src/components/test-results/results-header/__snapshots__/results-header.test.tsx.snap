// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsHeader component should match snapshot 1`] = `
<div
  className="resultsHeader"
>
  <div
    className="resultsHeaderIconWrapper"
  >
    <img
      alt="test result icon"
      className="resultsHeaderIcon"
      src="test-results-light-blue.svg"
    />
  </div>
  <div
    className="resultsHeaderTitles"
  >
    <span
      className="resultsHeaderTitlesMain"
    >
      Comprehensive Metabolic Panel
    </span>
    <span
      className="resultsHeaderTitlesAdditional"
    >
      Sat, Aug 14, 2021
    </span>
  </div>
</div>
`;

exports[`ResultsHeader should work with empty data 1`] = `
<div
  className="resultsHeader"
>
  <div
    className="resultsHeaderIconWrapper"
  >
    <img
      alt="test result icon"
      className="resultsHeaderIcon"
      src="test-results-light-blue.svg"
    />
  </div>
  <div
    className="resultsHeaderTitles"
  >
    <span
      className="resultsHeaderTitlesMain"
    />
    <span
      className="resultsHeaderTitlesAdditional"
    />
  </div>
</div>
`;
